pyqtgraph/graphicsItems/AxisItem.py

```diff
@@ -113,8 +128,9 @@ def __init__(self, orientation, pen=None, textPen=None, linkView=None, parent=No
             self._linkToView_internal(linkView)
 
         self.grid = False
-        
-        #self.setCacheMode(self.DeviceCoordinateCache)
+
+        if lru_cache_tick_strings:
+            self.tickStrings = functools.lru_cache(maxsize=2**20)(self.tickStrings)
```

Comment:
didn't know you could declare it this way without the decorator, neat!

Reply:
Yah that's what `@decorator` is transformed to iirc.

Reply:
Yah, check [the docs](https://docs.python.org/3.10/glossary.html#term-decorator)